### ISSUE TITLE
[BUG] fold pad(constant_value=0) into maxpool

### ISSUE THREAD

**Issue by xieyi4650** [daquexian/onnx-simplifier#290]
**Describe the bug**
fold pad(constant_value=0) into maxpool, 
but should fold fold pad(constant_value=-inf) into maxpool

![image](https://github.com/daquexian/onnx-simplifier/assets/23718469/2e716e8f-5bb4-469e-9915-193f3fb56938)
![image](https://github.com/daquexian/onnx-simplifier/assets/23718469/4459bee6-fbd2-43f0-b14d-90a48c2be1d3)

**Comment by take-cheeze** [comment]
@xieyi4650 
Seems you are right about it for MaxPool.
Applying the pass to AveragePool seems to be OK.
Though the actual code seems to live in onnx optimizer:
https://github.com/onnx/optimizer/blob/5627f51fc949415ef09db3cf9c7d376b5698a93d/onnxoptimizer/passes/fuse_pad_into_pool.h#L83

**Comment by take-cheeze** [comment]
Confirmed as a real correctness bug on onnxsim 0.6.5. A constant `Pad` with value `0` folded into a following `MaxPool` changes the result, because MaxPool padding uses `-inf` semantics, not `0`:

```python
# Pad(mode=constant, value=0, pads=[...,1,1,...,1,1]) -> MaxPool(k=3, s=3)
# onnxsim folds Pad into MaxPool.pads and drops the Pad node.
# With an all-negative input (ORT optimizations disabled):
orig  : [0. 0. 0. 0.]                                  # 0-padding wins the max
simpl : [-0.097 -0.317 -0.466 -0.450]                  # no padding -> different result
allclose: False
```

So the fold is only valid when the pad value is `-inf` (or the input is provably non-negative). This lives in the onnx-optimizer `fuse_pad_into_pool` pass (C++), so the fix belongs there: it must not fuse a constant `Pad` into `MaxPool` unless the pad value is `-inf`. Leaving this open to track that.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by take-cheeze** [comment]
This is fixed by #479, which bumps the `onnx-optimizer` submodule to a branch containing the `fuse_pad_into_pool` fix.

MaxPool ignores its padded elements, which is equivalent to padding with `-inf` — not `0`. The pass previously folded a `Pad` with `constant_value=0` (or an unspecified value, which defaults to `0`) into MaxPool, which changes the result whenever a pooling window's real values are all negative: `Pad(0) + MaxPool` yields `0`, while the fused MaxPool yields the (negative) window maximum.

The required `Pad` constant value now depends on the pool type — `0` for AveragePool (with `count_include_pad=1`) and `-inf` for MaxPool — and an unspecified `Pad` value correctly blocks fusion into MaxPool instead of being treated as a match. Tests were updated to assert no fusion for `value=0`/default and to confirm `value=-inf` folds correctly.

Will close once #479 is merged.

---
_Generated by [Claude Code](https://claude.ai/code)_
